Cargill/splinter
# Add Biome as auth provider for REST API

Adds Biome credentials as an option for auth in the Splinter REST API

To test:

Run splinterd with experimental features
```bash
CARGO_ARGS="-- --features=experimental" docker-compose up --build
```

Verify that the REST API is checking auth by checking that this command gets a 401 response
```bash
curl -I http://localhost:8080/status
```

Registry a Biome user
```bash
curl --request POST \
  --header "Content-Type: application/json" \
  --data '{"username":"test","hashed_password":"123"}' \
  http://localhost:8080/biome/register
```

Login the Biome user to get a Biome JWT
```bash
curl --request POST \
  --header "Content-Type: application/json" \
  --data '{"username":"test","hashed_password":"123"}' \
  http://localhost:8080/biome/login
```

Copy the Biome JWT (the "token" field from the previous command) and verify that it can be used to get the node's status
```bash
curl --header "Authorization: Bearer <biome-token>" http://localhost:8080/status
```
